### ISSUE TITLE
Octokit.AuthorizationException in dotneteng-status

### ISSUE THREAD

**Issue by riarenas** [dotnet/dnceng#4405]
https://dev.azure.com/dnceng/internal/_build/results?buildId=2575453&view=results didn't open an issue. Investigate

### Release Note Category
- [ ] Feature changes/additions 
- [ ] Bug fixes
- [ ] Internal Infrastructure Improvements
### Release Note Description

**Comment by riarenas** [comment]
I wasn't able to make much progress here. The only secret we've had to cycle recently is in https://dnceng.visualstudio.com/internal/_git/dotnet-helix-service?path=/.vault-config/helixprodservice.yaml&version=GBmain&line=170&lineEnd=177&lineStartColumn=1&lineEndColumn=1&lineStyle=plain&_a=contents. The secret specifies only the private key should be cycled, so that's what I've been doing. 

It's possible that the secret was cycled successfully and this isn't related, but that particular secret secret seemed to be used for this particular endpoint so there might still be something there.

**Comment by premun** [comment]
![Image](https://github.com/user-attachments/assets/f80a29cf-ede4-4274-b3c2-dfcf6ebb77b8)

It does not repro always though

**Comment by MilenaHristova** [comment]
I think I found a clue
If we also summarize by operation_Name ( [query](https://ms.portal.azure.com#@72f988bf-86f1-41af-91ab-2d7cd011db47/blade/Microsoft_OperationsManagementSuite_Workspace/Logs.ReactView/resourceId/%2Fsubscriptions%2F68672ab8-de0c-40f1-8d1b-ffb20bd62c0f%2FresourceGroups%2Fmonitoring%2Fproviders%2Fmicrosoft.insights%2Fcomponents%2FDotNetEng-Status-Prod/source/LogsBlade.AnalyticsShareLinkToQuery/q/H4sIAAAAAAAAAytKLSxNLS4p5qpRKC7NzU0syqxKVUjOL80r0dBUSKpUKEotLs0pcc5PSdVRKC3K0VHIL0gtSizJzM%252BL90vMTQUAjFNIGj8AAAA%253D/timespan/2024-11-10T14%3A46%3A10.409Z%2F2024-11-11T14%3A46%3A10.409Z)) most of the failed requests are for operation `GitHubHook/IssuesHook`
Looking at which GitHub issues are mentioned in the traces
All the failed requests are processing issues in dnceng - [query](https://ms.portal.azure.com#@72f988bf-86f1-41af-91ab-2d7cd011db47/blade/Microsoft_OperationsManagementSuite_Workspace/Logs.ReactView/resourceId/%2Fsubscriptions%2F68672ab8-de0c-40f1-8d1b-ffb20bd62c0f%2FresourceGroups%2Fmonitoring%2Fproviders%2Fmicrosoft.insights%2Fcomponents%2FDotNetEng-Status-Prod/source/LogsBlade.AnalyticsShareLinkToQuery/q/H4sIAAAAAAAAA03NPQ7CMAwF4J1TPGWpxNITdIV2QdwAmcS04ceGOBELh6ehUsVm6b33OfGrsGXD5oP3xIlhxXs2Q9fB7ehu7NZIn5woR5XTgR6MiQxuH3Nfzr3qrR3MZquernJXjYKcaNag8jcewipuF%252BSYtP6MMiL%252BEJCvVTQcYubQ4KJpiRDEs4yt%252BwJSNoL3ugAAAA%253D%253D/timespan/P1D)
The successful requests are for other repos - [query](https://ms.portal.azure.com#@72f988bf-86f1-41af-91ab-2d7cd011db47/blade/Microsoft_OperationsManagementSuite_Workspace/Logs.ReactView/resourceId/%2Fsubscriptions%2F68672ab8-de0c-40f1-8d1b-ffb20bd62c0f%2FresourceGroups%2Fmonitoring%2Fproviders%2Fmicrosoft.insights%2Fcomponents%2FDotNetEng-Status-Prod/source/LogsBlade.AnalyticsShareLinkToQuery/q/H4sIAAAAAAAAA03NsQ7CMAwE0J2vOGWpxMIXdKZdEAM7ShOXGkQMdiIWPp6GSlU3S3f3rPQuZNmw%252B%252BIzkRKshEBmaFu4ixZyayIvUp9Z0vXkn4TJG9yRc1eGTuRx6M1mqp6uanfhhKx%252BxiBpM%252B7jKu4X5KxSX3K6gf8IfKhVNBQ5U2wwii6R%252BwFTIAOPsQAAAA%253D%253D/timespan/P1D)

**Comment by MilenaHristova** [comment]
@dkurepa updated Maestro recently to use `Octokit.Webhooks.AspNetCore` - https://github.com/dotnet/arcade-services/pull/4111
@riarenas would you consider doing the same change in dnceng repo, we think it will help

**Comment by riarenas** [comment]
Whatever fixes the issue sounds good to me

**Comment by riarenas** [comment]
Could you summarize how the move to the new library would help with these exceptions?

**Comment by riarenas** [comment]
https://github.com/dotnet/dnceng/issues/4299#issuecomment-2474336930 was posted today from the staging enviroment, which makes me question why a library update would be needed to solve this.

Are we sure it's not just a production secret that isn't working properly?

**Comment by MilenaHristova** [comment]
This is the issue opened in arcade-services mentioning that the current library is deprecated https://github.com/dotnet/arcade-services/issues/3527
I'd need more time to understand the change in authentication, looks like in Maestro there is a different secret used instead of installation and private key
The library handles payload parsing and integration with webhook events  - one of our theories is that the service might be getting incorrect payload

**Comment by riarenas** [comment]
I used secret manager to cycle https://github.com/dotnet/dnceng/blob/main/.vault-config/shared/dotneteng-status-secrets.yaml#L1 and restarted the service. 

I verified the alerting functionality is working again in https://github.com/dotnet/dnceng/issues/4449 and that the issue mirroring from azure devops to github is working in https://github.com/dotnet/dnceng/issues/4450. 

I have a theory on what happened and I will add more details after I check that out tomorrow.

**Comment by riarenas** [comment]
When cycling the secret used in https://github.com/dotnet/dnceng/issues/4405#issuecomment-2464853606 and the secret used in https://github.com/dotnet/dnceng/blob/main/.vault-config/shared/dotneteng-status-secrets.yaml#L1 both mention that older private keys should be deleted, but both secrets need to have an up to date private key. I deleted one key too many when I cycled the helix service secret.
